internal/api/hcpopenshiftclusterexternalauth.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2025 Microsoft Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	validator "github.com/go-playground/validator/v10"
+
+	"github.com/Azure/ARO-HCP/internal/api/arm"
+)
+
+// HCPOpenShiftClusterExternalAuth represents the external auth config resource for ARO HCP
+// OpenShift clusters.
+type HCPOpenShiftClusterExternalAuth struct {
+	arm.ProxyResource
+	Properties HCPOpenShiftClusterExternalAuthProperties `json:"properties,omitempty" validate:"required_for_put"`
+}
+
+// HCPOpenShiftClusterNodePoolProperties represents the property bag of a
+// HCPOpenShiftClusterNodePool resource.
+type HCPOpenShiftClusterExternalAuthProperties struct {
+	ProvisioningState arm.ExternalAuthProvisioningState `json:"provisioningState,omitempty"       visibility:"read"`
+	Condition         ExternalAuthCondition             `json:"condition,omitzero"                visibility:"read"`
+	Issuer            TokenIssuerProfile                `json:"issuer"                            visibility:"read create update"       validate:"required"`
+	Clients           []ExternalAuthClientProfile       `json:"clients,omitempty"                 visibility:"read create update"       validate:"max=20"`
+	Claim             ExternalAuthClaimProfile          `json:"claim"                             visibility:"read create update"       validate:"required"`
+}
+
+/** Condition defines an observation of the external auth state. */
+type ExternalAuthCondition struct {
+	ConditionType      ExternalAuthConditionType `json:"type"                         visibility:"read"     validate:"omitempty,enum_externalauthconditiontype"`
+	Status             ConditionStatusType       `json:"status"                       visibility:"read"     validate:"omitempty,enum_externalauthconditionstatustype"`
+	LastTransitionTime time.Time                 `json:"lastTransitionTime"           visibility:"read"`
+	Reason             string                    `json:"reason"                       visibility:"read"`
+	Message            string                    `json:"message"                      visibility:"read"`
```

Comment:
You could omit the `visibility` tags here if you wanted to.  They will all inherit `visibility:"read"` from the `Condition` field above.

Same goes for the other structs below, it looks like.